opentelemetry-resourcedetector-gcp/tests/test_gcp_resource_detector.py

```diff
@@ -95,20 +95,16 @@ def pop_environ_key(key):
         os.environ.pop(key)
 
 
-def clear_gke_env_vars():
```

Comment:
This file is the test for the existing resource detector which is being replaced.

It was leaking changes to `os.environ` into the other tests which I had to fix